autobot-backend/middleware/idempotency_middleware_test.py

```
@@ -0,0 +1,291 @@
+class TestConcurrentReplay:
+    def test_an_in_flight_key_is_refused_rather_than_raced(self, app_and_calls):
+        """Two retries arriving together must not both create.
+
+        The claim is held directly here, which is what a request still executing
+        looks like to the second caller — the state a two-state store would
+        collapse into "unseen" and let through.
+        """
+        app, calls, redis = app_and_calls
+        key = storage_key("anonymous", "POST", "/create", "k1")
+
+        async def _hold_the_claim():
```

Ruff (0.16.3)

[warning] Missing return type annotation for private function `_hold_the_claim`

Add return type annotation: `None`

(ANN202)